### PR TITLE
Test: Adding e2e tests, test README, and pipeline infra for tests

### PR DESCRIPTION
**Pull request by gracewehner**
### PR Process
- Add pull request template, including a checklist for testing
- Add a semantic conventional commit check for PR titles (for creating the changelog for releases)

### Pipeline
- Split pipeline into two stages: `Build` (images and charts) and `Deploy` (on CI/CD clusters)
- Add an environment lock to the `Deploy` stage so that only one merge to main can acquire that lock at a time
- Add running the TestKube tests on the CI/CD clusters to the `Deploy` stage, so that a new version cannot be deployed on the CI/CD clusters until the current tests have finished running

### Docs
- Add a testing README for processes, Ginkgo and TestKube info, how to run the tests and bootstrap clusters, how to add new tests, etc.

### Tests
- Add the `otelcollector/tests` folder that has the starting e2e tests, split up into test suites

### CODE REVIEW

**Review by vishiy — APPROVED** [review]

File: .github/semantic.yml [hunk 1]
```diff
@@ -0,0 +1,17 @@
...
+  - fix
+  - docs
+  - style
+  - refactor
+  - perf
+  - test
+  - build
+  - ci
+  - chore
+  - revert
```
Comment (gracewehner) [inline .github/semantic.yml:17]: For doc PRs, don't run all checks

File: .pipelines/azure-pipeline-build.yml [hunk 2]
```diff
@@ -24,1062 +24,1122 @@ variables:
...
+              kubernetesCluster: 'ci-dev-aks-mac-eus'
+              namespace: 'default'
+              command: 'upgrade'
+              chartType: 'FilePath'
+              chartPath: '$(Build.SourcesDirectory)/otelcollector/deploy/addon-chart/azure-monitor-metrics-addon/'
+              releaseName: 'ama-metrics'
+              waitForExecution: false
+              arguments: --dependency-update --values $(Build.SourcesDirectory)/otelcollector/deploy/addon-chart/azure-monitor-metrics-addon/values.yaml
+
+  - deployment: Testkube
```
Comment (gracewehner) [inline .pipelines/azure-pipeline-build.yml:1090]: Re-run with only running the tests

Comment (gracewehner) [reply .pipelines/azure-pipeline-build.yml:1090]: Disable whole stage re-runs possible?

File: otelcollector/test/containerstatus/container_status_test.go [hunk 3]
```diff
@@ -0,0 +1,79 @@
+package containerstatus
+
+import (
+	"prometheus-collector/otelcollector/test/utils"
+
+	. "github.com/onsi/ginkgo/v2"
+	. "github.com/onsi/gomega"
+)
+
+/*
```
Comment (gracewehner) [inline otelcollector/test/containerstatus/container_status_test.go:10]: New version addon-token-adapter?

File: otelcollector/test/prometheusui/prometheus_ui_test.go [hunk 4]
```diff
@@ -0,0 +1,161 @@
...
+    for _, target := range targetsResult.Active {
+      Expect(target.DiscoveredLabels).NotTo(BeNil())
+      Expect(target.Labels).NotTo(BeNil())
+    }
+    Expect(targetsResult.Dropped).NotTo(BeNil())
+  },
+  Entry("when called inside ama-metrics replica pod", "kube-system", "rsName", "ama-metrics", "prometheus-collector"),
+  Entry("when called inside the ama-metrics-node pod", "kube-system", "dsName", "ama-metrics-node", "prometheus-collector"),
+)
+
```
Comment (gracewehner) [inline otelcollector/test/prometheusui/prometheus_ui_test.go:98]: Add check for /metrics

File: .pipelines/azure-pipeline-build.yml [hunk 5]
```diff
@@ -2,12 +2,50 @@ trigger:
...
+    - 'ArcArmTemplate/*'
+    - 'ArcBicepTemplate/*'
+    - 'Azure-ARM-templates/*'
+    - 'GeneratedMonitoringArtifacts/*'
+    - 'internal/alerts/*'
+    - 'internal/docs/*'
+    - 'internal/monitoring/*'
+    - 'internal/scripts/*'
+    - 'internal/mixins/*'
+    - 'tools/*'
```
Comment (gracewehner) [inline .pipelines/azure-pipeline-build.yml:23]: Make sure with separate branch PR and this file

File: otelcollector/test/test-cluster-yamls/configmaps/ama-metrics-settings-configmap.yaml [hunk 6]
```diff
@@ -0,0 +1,67 @@
+kind: ConfigMap
```
Comment (gracewehner) [inline otelcollector/test/test-cluster-yamls/configmaps/ama-metrics-settings-configmap.yaml:1]: Test labels for not using configmaps

File: otelcollector/test/prometheusui/suite_test.go [hunk 7]
```diff
@@ -0,0 +1,35 @@
...
+var K8sClient 	*kubernetes.Clientset
+var Cfg       	*rest.Config
+
+func TestPrometheusUI(t *testing.T) {
+  RegisterFailHandler(Fail)
+
+  RunSpecs(t, "Prometheus UI Test Suite")
+}
+
+var _ = BeforeSuite(func() {
```
Comment (gracewehner) [inline otelcollector/test/prometheusui/suite_test.go:27]: Later: could we apply configmaps before?

**Review by bragi92 — DISMISSED** [review]

**Review by vishiy — APPROVED** [review]

**Review by bragi92 — APPROVED** [review]

**Review by vishiy — APPROVED** [review]

**Pull request merged**